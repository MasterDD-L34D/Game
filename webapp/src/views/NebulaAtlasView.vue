<template>
  <section class="nebula-atlas-view">
    <NebulaProgressModule
      :header="moduleState.header"
      :cards="moduleState.cards"
      :timeline-entries="moduleState.timelineEntries"
      :evolution-matrix="moduleState.evolutionMatrix"
      :share="moduleState.share"
    />

    <section class="nebula-atlas-view__live" aria-live="polite">
      <header class="nebula-atlas-view__live-header">
        <div>
          <h3>Telemetria live</h3>
          <p>Indicatori dal generatore Nebula combinati con il dataset atlas.</p>
        </div>
        <div class="nebula-atlas-view__status">
          <span v-if="moduleState.loading" class="nebula-atlas-view__badge">Aggiornamento…</span>
          <span v-else class="nebula-atlas-view__badge" :data-tone="statusTone">{{ statusLabel }}</span>
          <small>Ultimo sync: {{ moduleState.telemetrySummary.updatedAt ? moduleState.telemetrySummary.updatedAt : '—' }}</small>
        </div>
      </header>

      <div v-if="moduleState.error" class="nebula-atlas-view__error" role="status">
        {{ moduleState.error.message }}
      </div>

      <div class="nebula-atlas-view__grid" v-else>
        <div class="nebula-atlas-view__metrics">
          <article v-for="card in summaryCards" :key="card.id" class="nebula-atlas-view__metric" :data-tone="card.tone">
            <h4>{{ card.label }}</h4>
            <strong>{{ card.value }}</strong>
          </article>
        </div>

        <div class="nebula-atlas-view__readiness">
          <h4>Readiness branchi</h4>
          <ul>
            <li v-for="chip in readinessChips" :key="chip.id" :data-tone="chip.tone">
              <span>{{ chip.label }}</span>
              <strong>{{ chip.value }}</strong>
            </li>
          </ul>
        </div>

        <div class="nebula-atlas-view__chart" aria-label="Copertura QA">
          <header>
            <h4>Copertura QA media</h4>
            <span>{{ moduleState.telemetryCoverageAverage }}%</span>
          </header>
          <SparklineChart :points="coverageStream" color="#61d5ff" />
        </div>

        <div class="nebula-atlas-view__chart" aria-label="Incidenti telemetria">
          <header>
            <h4>Incidenti ultimi 7 giorni</h4>
            <span>{{ incidentStream.reduce((acc, value) => acc + value, 0) }}</span>
          </header>
          <SparklineChart :points="incidentStream" color="#ff6982" />
        </div>

        <div class="nebula-atlas-view__chart" aria-label="High priority">
          <header>
            <h4>High priority</h4>
            <span>{{ highPriorityStream.reduce((acc, value) => acc + value, 0) }}</span>
          </header>
          <SparklineChart :points="highPriorityStream" color="#f4c060" />
        </div>
      </div>

      <footer class="nebula-atlas-view__footer">
        <span>{{ moduleState.telemetrySummary.lastEventLabel }}</span>
        <button type="button" class="nebula-atlas-view__refresh" @click="refresh">Aggiorna ora</button>
      </footer>
    </section>
  </section>
</template>

<script setup>
import { computed } from 'vue';
import NebulaProgressModule from '../components/flow/NebulaProgressModule.vue';
import SparklineChart from '../components/shared/SparklineChart.vue';
import { useNebulaProgressModule } from '../modules/useNebulaProgressModule';

const moduleState = useNebulaProgressModule();

const summaryCards = computed(() => {
  const summary = moduleState.telemetrySummary.value;
  return [
    { id: 'total', label: 'Eventi totali', value: summary.total, tone: 'neutral' },
    { id: 'open', label: 'Eventi aperti', value: summary.open, tone: summary.open > 0 ? 'warning' : 'success' },
    { id: 'high', label: 'Priorità alta', value: summary.highPriority, tone: summary.highPriority > 0 ? 'critical' : 'neutral' },
    { id: 'ack', label: 'Acknowledged', value: summary.acknowledged, tone: 'success' },
  ];
});

const readinessChips = computed(() => {
  const distribution = moduleState.telemetryDistribution.value;
  return [
    { id: 'success', label: 'Pronte', tone: 'success', value: distribution.success },
    { id: 'warning', label: 'In attesa', tone: 'warning', value: distribution.warning },
    { id: 'neutral', label: 'Neutrali', tone: 'neutral', value: distribution.neutral },
    { id: 'critical', label: 'Bloccate', tone: 'critical', value: distribution.critical },
  ];
});

const coverageStream = computed(() => moduleState.telemetryStreams.value.coverage);
const incidentStream = computed(() => moduleState.telemetryStreams.value.incidents);
const highPriorityStream = computed(() => moduleState.telemetryStreams.value.highPriority);

const statusTone = computed(() => (moduleState.telemetrySummary.value.open > 0 ? 'warning' : 'success'));
const statusLabel = computed(() =>
  moduleState.loading.value ? 'Aggiornamento…' : moduleState.telemetrySummary.value.lastEventLabel,
);

const refresh = () => moduleState.refresh();
</script>

<style scoped>
.nebula-atlas-view {
  display: grid;
  gap: 2.5rem;
}

.nebula-atlas-view__live {
  background: rgba(5, 8, 13, 0.9);
  border: 1px solid rgba(97, 213, 255, 0.35);
  border-radius: 20px;
  padding: 2rem;
  color: #e6f3ff;
  display: grid;
  gap: 2rem;
}

.nebula-atlas-view__live-header {
  display: flex;
  justify-content: space-between;
  gap: 1.5rem;
  flex-wrap: wrap;
}

.nebula-atlas-view__status {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  gap: 0.4rem;
}

.nebula-atlas-view__badge {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  border-radius: 999px;
  padding: 0.25rem 0.85rem;
  font-size: 0.85rem;
  background: rgba(97, 213, 255, 0.18);
  color: #61d5ff;
}

.nebula-atlas-view__badge[data-tone='warning'] {
  background: rgba(244, 192, 96, 0.18);
  color: #f4c060;
}

.nebula-atlas-view__badge[data-tone='success'] {
  background: rgba(115, 255, 206, 0.18);
  color: #73ffce;
}

.nebula-atlas-view__grid {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(240px, 1fr));
  gap: 1.5rem;
}

.nebula-atlas-view__metrics {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(140px, 1fr));
  gap: 1rem;
}

.nebula-atlas-view__metric {
  padding: 1.25rem;
  border-radius: 16px;
  background: rgba(15, 23, 42, 0.65);
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
  border: 1px solid rgba(97, 213, 255, 0.18);
}

.nebula-atlas-view__metric[data-tone='warning'] {
  border-color: rgba(244, 192, 96, 0.35);
}

.nebula-atlas-view__metric[data-tone='critical'] {
  border-color: rgba(255, 105, 130, 0.4);
}

.nebula-atlas-view__metric[data-tone='success'] {
  border-color: rgba(115, 255, 206, 0.35);
}

.nebula-atlas-view__metric h4 {
  margin: 0;
  font-size: 0.85rem;
  text-transform: uppercase;
  letter-spacing: 0.08em;
  color: rgba(230, 243, 255, 0.75);
}

.nebula-atlas-view__metric strong {
  font-size: 1.8rem;
}

.nebula-atlas-view__readiness {
  padding: 1.5rem;
  border-radius: 16px;
  background: rgba(15, 23, 42, 0.55);
  border: 1px solid rgba(97, 213, 255, 0.18);
}

.nebula-atlas-view__readiness h4 {
  margin: 0 0 0.75rem;
  text-transform: uppercase;
  letter-spacing: 0.08em;
  font-size: 0.85rem;
  color: rgba(230, 243, 255, 0.75);
}

.nebula-atlas-view__readiness ul {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(120px, 1fr));
  gap: 0.75rem;
  list-style: none;
  margin: 0;
  padding: 0;
}

.nebula-atlas-view__readiness li {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0.75rem 1rem;
  border-radius: 12px;
  background: rgba(230, 243, 255, 0.05);
  border: 1px solid rgba(97, 213, 255, 0.12);
}

.nebula-atlas-view__readiness li[data-tone='warning'] {
  border-color: rgba(244, 192, 96, 0.25);
}

.nebula-atlas-view__readiness li[data-tone='critical'] {
  border-color: rgba(255, 105, 130, 0.35);
}

.nebula-atlas-view__readiness li[data-tone='success'] {
  border-color: rgba(115, 255, 206, 0.3);
}

.nebula-atlas-view__chart {
  padding: 1.5rem;
  border-radius: 16px;
  background: rgba(15, 23, 42, 0.55);
  border: 1px solid rgba(97, 213, 255, 0.18);
  display: grid;
  gap: 1rem;
}

.nebula-atlas-view__chart header {
  display: flex;
  justify-content: space-between;
  align-items: baseline;
  gap: 1rem;
}

.nebula-atlas-view__chart h4 {
  margin: 0;
  text-transform: uppercase;
  letter-spacing: 0.08em;
  font-size: 0.85rem;
  color: rgba(230, 243, 255, 0.75);
}

.nebula-atlas-view__chart span {
  font-weight: 600;
  color: rgba(230, 243, 255, 0.9);
}

.nebula-atlas-view__footer {
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 1rem;
  font-size: 0.9rem;
  color: rgba(230, 243, 255, 0.7);
}

.nebula-atlas-view__refresh {
  background: linear-gradient(135deg, #61d5ff, #9f7bff);
  border: none;
  color: #05080d;
  font-weight: 600;
  padding: 0.5rem 1.5rem;
  border-radius: 999px;
  cursor: pointer;
  transition: transform 0.18s ease, box-shadow 0.18s ease;
}

.nebula-atlas-view__refresh:hover {
  transform: translateY(-1px);
  box-shadow: 0 10px 24px rgba(97, 213, 255, 0.35);
}

.nebula-atlas-view__error {
  padding: 1rem 1.5rem;
  border-radius: 12px;
  background: rgba(255, 105, 130, 0.15);
  border: 1px solid rgba(255, 105, 130, 0.35);
  color: #ffd5de;
}
</style>
