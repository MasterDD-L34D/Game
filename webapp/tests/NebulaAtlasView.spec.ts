import { describe, expect, it, beforeEach, vi } from 'vitest';
import { computed, ref } from 'vue';
import { mount } from '@vue/test-utils';

import NebulaAtlasView from '../src/views/NebulaAtlasView.vue';

const refreshMock = vi.fn();
const activateMock = vi.fn();

vi.mock('../src/modules/useNebulaProgressModule', () => ({
  useNebulaProgressModule: () => ({
    header: computed(() => ({
      datasetId: 'atlas-demo',
      title: 'Nebula Atlas Demo',
      summary: 'Dataset statico per QA',
      releaseWindow: 'Q4 路 2035',
      curator: 'Team Nebula',
    })),
    cards: computed(() => []),
    timelineEntries: computed(() => []),
    evolutionMatrix: computed(() => []),
    share: computed(() => ({})),
    telemetrySummary: computed(() => ({
      total: 128,
      open: 4,
      acknowledged: 92,
      highPriority: 2,
      lastEventAt: null,
      lastEventLabel: 'Sync mock 15m fa',
      updatedAt: '2035-04-01T12:00:00Z',
      mode: 'mock',
      isDemo: true,
      sourceLabel: 'Telemetria offline 路 demo',
    })),
    telemetryStreams: computed(() => ({
      coverage: [60, 72, 88],
      incidents: [2, 1, 0],
      highPriority: [1, 1, 0],
    })),
    telemetryDistribution: computed(() => ({
      success: 8,
      warning: 2,
      neutral: 1,
      critical: 0,
    })),
    telemetryCoverageAverage: computed(() => 82),
    telemetryStatus: computed(() => ({
      mode: 'mock',
      offline: true,
      variant: 'demo',
      label: 'Telemetria offline 路 demo',
    })),
    generatorStatus: computed(() => ({
      status: 'mock',
      label: 'Generatore offline 路 demo',
      generatedAt: null,
      updatedAt: '2035-04-01T12:00:00Z',
      sourceLabel: 'Generator telemetry offline',
    })),
    generatorMetrics: computed(() => ({
      generationTimeMs: 512,
      speciesTotal: 18,
      enrichedSpecies: 12,
      eventTotal: 2,
      datasetSpeciesTotal: 20,
      coverageAverage: 82,
      coreTraits: 44,
      optionalTraits: 22,
      synergyTraits: 10,
      expectedCoreTraits: 28,
    })),
    generatorStreams: computed(() => ({
      generationTime: [280, 340, 420],
      species: [10, 14, 18],
      enriched: [6, 9, 12],
    })),
    loading: computed(() => false),
    error: computed(() => null),
    lastUpdated: ref('2035-04-01T12:00:00Z'),
    refresh: refreshMock,
    activateDemoTelemetry: activateMock,
  }),
}));

describe('NebulaAtlasView', () => {
  beforeEach(() => {
    refreshMock.mockClear();
    activateMock.mockClear();
  });

  function mountView() {
    return mount(NebulaAtlasView, {
      global: {
        stubs: {
          NebulaProgressModule: {
            template: '<div data-test="progress-stub"></div>',
          },
          SparklineChart: {
            template: '<div class="sparkline-chart"></div>',
          },
        },
      },
    });
  }

  it('mostra il banner demo e genera uno snapshot stabile', () => {
    const wrapper = mountView();
    expect(wrapper.find('.nebula-atlas-view__banner').exists()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('espone i controlli manuali di sincronizzazione', async () => {
    const wrapper = mountView();
    const buttons = wrapper.findAll('button');
    expect(buttons.length).toBeGreaterThan(0);

    for (const button of buttons) {
      await button.trigger('click');
    }

    expect(refreshMock).toHaveBeenCalledTimes(2);
    expect(activateMock).toHaveBeenCalledTimes(2);
  });
});
