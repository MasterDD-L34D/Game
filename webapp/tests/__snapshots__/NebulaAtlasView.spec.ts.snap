// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NebulaAtlasView > mostra il banner demo e genera uno snapshot stabile 1`] = `
"<section data-v-43e6e8ac="" class="nebula-atlas-view">
  <aside data-v-43e6e8ac="" class="nebula-atlas-view__banner" role="status">
    <div data-v-43e6e8ac="">
      <h2 data-v-43e6e8ac="">Modalità demo attiva</h2>
      <p data-v-43e6e8ac=""> Il dataset Nebula Atlas Demo è disponibile in modalità offline coordinata e la console mantiene sincronizzazioni di supporto per garantire la continuità operativa. </p>
      <ul data-v-43e6e8ac="" class="nebula-atlas-view__banner-meta">
        <li data-v-43e6e8ac=""><span data-v-43e6e8ac="">Dataset</span><strong data-v-43e6e8ac="">Dataset offline · fallback</strong></li>
        <li data-v-43e6e8ac=""><span data-v-43e6e8ac="">Telemetria</span><strong data-v-43e6e8ac="">Telemetria offline · demo</strong></li>
      </ul>
    </div>
    <div data-v-43e6e8ac="" class="nebula-atlas-view__banner-actions"><button data-v-43e6e8ac="" type="button" class="nebula-atlas-view__banner-button"> Riprova sincronizzazione </button><button data-v-43e6e8ac="" type="button" class="nebula-atlas-view__banner-button"> Forza telemetria demo </button></div>
  </aside>
  <div data-v-43e6e8ac="" data-test="progress-stub" header="[object Object]" cards="[object Object]" timeline-entries="[object Object]" evolution-matrix="[object Object]" share="[object Object]" telemetry-status="[object Object]"></div>
  <section data-v-43e6e8ac="" class="nebula-atlas-view__live" aria-live="polite">
    <header data-v-43e6e8ac="" class="nebula-atlas-view__live-header">
      <div data-v-43e6e8ac="">
        <h3 data-v-43e6e8ac="">Telemetria live</h3>
        <p data-v-43e6e8ac="">Indicatori dal generatore Nebula combinati con il dataset atlas.</p>
      </div>
      <div data-v-43e6e8ac="" class="nebula-atlas-view__status"><span data-v-43e6e8ac="" class="nebula-atlas-view__badge" data-tone="offline">Telemetria offline · demo</span><small data-v-43e6e8ac="">Ultimo sync: 2035-04-01T12:00:00Z · Dataset offline · fallback · Telemetria offline · demo</small></div>
    </header>
    <div data-v-43e6e8ac="" class="nebula-atlas-view__grid">
      <div data-v-43e6e8ac="" class="nebula-atlas-view__metrics">
        <article data-v-43e6e8ac="" class="nebula-atlas-view__metric" data-tone="neutral">
          <h4 data-v-43e6e8ac="">Eventi totali</h4><strong data-v-43e6e8ac="">128</strong>
        </article>
        <article data-v-43e6e8ac="" class="nebula-atlas-view__metric" data-tone="warning">
          <h4 data-v-43e6e8ac="">Eventi aperti</h4><strong data-v-43e6e8ac="">4</strong>
        </article>
        <article data-v-43e6e8ac="" class="nebula-atlas-view__metric" data-tone="critical">
          <h4 data-v-43e6e8ac="">Priorità alta</h4><strong data-v-43e6e8ac="">2</strong>
        </article>
        <article data-v-43e6e8ac="" class="nebula-atlas-view__metric" data-tone="success">
          <h4 data-v-43e6e8ac="">Acknowledged</h4><strong data-v-43e6e8ac="">92</strong>
        </article>
      </div>
      <div data-v-43e6e8ac="" class="nebula-atlas-view__readiness">
        <h4 data-v-43e6e8ac="">Readiness branchi</h4>
        <ul data-v-43e6e8ac="">
          <li data-v-43e6e8ac="" data-tone="success" data-mode="demo">
            <div data-v-43e6e8ac="" class="nebula-atlas-view__chip-label"><span data-v-43e6e8ac="">Pronte</span><small data-v-43e6e8ac="">Telemetria demo</small></div><strong data-v-43e6e8ac="">8</strong>
          </li>
          <li data-v-43e6e8ac="" data-tone="warning" data-mode="demo">
            <div data-v-43e6e8ac="" class="nebula-atlas-view__chip-label"><span data-v-43e6e8ac="">In attesa</span><small data-v-43e6e8ac="">Telemetria demo</small></div><strong data-v-43e6e8ac="">2</strong>
          </li>
          <li data-v-43e6e8ac="" data-tone="neutral" data-mode="demo">
            <div data-v-43e6e8ac="" class="nebula-atlas-view__chip-label"><span data-v-43e6e8ac="">Neutrali</span><small data-v-43e6e8ac="">Telemetria demo</small></div><strong data-v-43e6e8ac="">1</strong>
          </li>
          <li data-v-43e6e8ac="" data-tone="critical" data-mode="demo">
            <div data-v-43e6e8ac="" class="nebula-atlas-view__chip-label"><span data-v-43e6e8ac="">Bloccate</span><small data-v-43e6e8ac="">Telemetria demo</small></div><strong data-v-43e6e8ac="">0</strong>
          </li>
        </ul>
      </div>
      <div data-v-43e6e8ac="" class="nebula-atlas-view__chart" aria-label="Copertura QA">
        <header data-v-43e6e8ac="">
          <h4 data-v-43e6e8ac="">Copertura QA media</h4><span data-v-43e6e8ac="">82%</span>
        </header>
        <div data-v-43e6e8ac="" class="sparkline-chart" points="60,72,88" color="#61d5ff" variant="demo"></div>
      </div>
      <div data-v-43e6e8ac="" class="nebula-atlas-view__chart" aria-label="Incidenti telemetria">
        <header data-v-43e6e8ac="">
          <h4 data-v-43e6e8ac="">Incidenti ultimi 7 giorni</h4><span data-v-43e6e8ac="">3</span>
        </header>
        <div data-v-43e6e8ac="" class="sparkline-chart" points="2,1,0" color="#ff6982" variant="demo"></div>
      </div>
      <div data-v-43e6e8ac="" class="nebula-atlas-view__chart" aria-label="High priority">
        <header data-v-43e6e8ac="">
          <h4 data-v-43e6e8ac="">High priority</h4><span data-v-43e6e8ac="">2</span>
        </header>
        <div data-v-43e6e8ac="" class="sparkline-chart" points="1,1,0" color="#f4c060" variant="demo"></div>
      </div>
      <div data-v-43e6e8ac="" class="nebula-atlas-view__generator" aria-label="Telemetria generatore">
        <header data-v-43e6e8ac="">
          <div data-v-43e6e8ac="">
            <h4 data-v-43e6e8ac="">Generatore Nebula</h4><small data-v-43e6e8ac="">Generator telemetry offline</small>
          </div><span data-v-43e6e8ac="" class="nebula-atlas-view__badge" data-tone="offline">Generatore offline · demo</span>
        </header>
        <p data-v-43e6e8ac="" class="nebula-atlas-view__generator-meta">Ultima run: 01/04/2035, 12:00:00</p>
        <div data-v-43e6e8ac="" class="nebula-atlas-view__metrics nebula-atlas-view__metrics--generator">
          <article data-v-43e6e8ac="" class="nebula-atlas-view__metric" data-tone="neutral">
            <h4 data-v-43e6e8ac="">Specie totali</h4><strong data-v-43e6e8ac="">18</strong>
            <!--v-if-->
          </article>
          <article data-v-43e6e8ac="" class="nebula-atlas-view__metric" data-tone="warning">
            <h4 data-v-43e6e8ac="">Blueprint arricchiti</h4><strong data-v-43e6e8ac="">12</strong><small data-v-43e6e8ac="">Target 20</small>
          </article>
          <article data-v-43e6e8ac="" class="nebula-atlas-view__metric" data-tone="warning">
            <h4 data-v-43e6e8ac="">Eventi runtime</h4><strong data-v-43e6e8ac="">2</strong>
            <!--v-if-->
          </article>
          <article data-v-43e6e8ac="" class="nebula-atlas-view__metric" data-tone="success">
            <h4 data-v-43e6e8ac="">Copertura atlas</h4><strong data-v-43e6e8ac="">82%</strong>
            <!--v-if-->
          </article>
        </div>
        <div data-v-43e6e8ac="" class="nebula-atlas-view__generator-charts">
          <div data-v-43e6e8ac="" class="nebula-atlas-view__chart" aria-label="Tempo generazione">
            <header data-v-43e6e8ac="">
              <h4 data-v-43e6e8ac="">Tempo generazione</h4><span data-v-43e6e8ac="">512 ms</span>
            </header>
            <div data-v-43e6e8ac="" class="sparkline-chart" points="280,340,420" color="#7c5cff" variant="demo"></div>
          </div>
          <div data-v-43e6e8ac="" class="nebula-atlas-view__chart" aria-label="Specie generate">
            <header data-v-43e6e8ac="">
              <h4 data-v-43e6e8ac="">Specie generate</h4><span data-v-43e6e8ac="">18</span>
            </header>
            <div data-v-43e6e8ac="" class="sparkline-chart" points="10,14,18" color="#4ade80" variant="demo"></div>
          </div>
          <div data-v-43e6e8ac="" class="nebula-atlas-view__chart" aria-label="Blueprint arricchiti">
            <header data-v-43e6e8ac="">
              <h4 data-v-43e6e8ac="">Blueprint arricchiti</h4><span data-v-43e6e8ac="">12</span>
            </header>
            <div data-v-43e6e8ac="" class="sparkline-chart" points="6,9,12" color="#ff99cc" variant="demo"></div>
          </div>
        </div>
      </div>
    </div>
    <footer data-v-43e6e8ac="" class="nebula-atlas-view__footer"><span data-v-43e6e8ac="">Sync mock 15m fa</span>
      <div data-v-43e6e8ac="" class="nebula-atlas-view__controls"><button data-v-43e6e8ac="" type="button" class="nebula-atlas-view__refresh">Aggiorna ora</button><button data-v-43e6e8ac="" type="button" class="nebula-atlas-view__refresh">Carica mock</button></div>
    </footer>
  </section>
</section>"
`;
